### PR TITLE
Fix type mismatch error, add tests for negative prompts

### PR DESCRIPTION
**Pull request by anton-l**
* Makes sure that `prompts` are not wrapped in a list too soon (causing `type(prompt) is not type(negative_prompt)`)
* Adds test for negative prompts in SD pipelines

Fixes https://github.com/huggingface/diffusers/issues/820

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py [hunk 1]
```diff
@@ -195,7 +195,6 @@ def __call__(
         """
         if isinstance(prompt, str):
             batch_size = 1
-            prompt = [prompt]
```
Comment (anton-l) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:198]: the cause of the error in https://github.com/huggingface/diffusers/issues/820

**Review by patil-suraj — APPROVED** [review]
Looks good, thanks a lot for fixing this!
